metrics/src/lib.rs

```diff
@@ -32,46 +32,67 @@ pub fn install_metrics() {
 // function block. Such code is possible when async closures become stable.
 #[macro_export]
 macro_rules! record_duration {
-    ( $metric_name:literal, $e:expr ) => {{
+    ( $metric_name:expr, $e:expr ) => {{
```

Comment:
This is relaxed because `metrics` can also accept static `str` refs and owned `String` in addition to literals. Static refs is what I mainly needed it to accept in order to avoid duplicating counter calls everywhere.